app/javascript/mastodon/features/compose/components/emoji_picker_dropdown.js

```diff
@@ -212,6 +218,27 @@ class EmojiPickerMenu extends React.PureComponent {
     }
   }
 
+  getCustomEmojis () {
+    const emojis = [];
+
+    this.props.custom_emojis.forEach(emoji => {
+      const shortcode = emoji.get('shortcode');
+      const url = emoji.get('url');
+      const name = shortcode.replace(':', '');
+
+      emojis.push({
+        name,
+        short_names: [name],
+        text: '',
+        emoticons: [],
+        keywords: [name],
+        imageUrl: url,
+      });
+    });
+
+    return emojis;
```

Comment:
```javascript
return this.props.custom_emojis.map(emoji => {
  const shortcode = emoji.get('shortcode');
  const url = emoji.get('url');
  const name = shortcode.replace(':', '');

  return {
    name,
    short_names: [name],
    text: '',
    emoticons: [],
    keywords: [name],
    imageUrl: url,
  };
});
```

How about using Array.prototype.map as above?

Reply:
I could just use `.toJS()` on the Immutable.List, probably, likely without any performance losses. But it's not a big deal...

Reply:
i agree